main.go

```diff
@@ -280,49 +278,41 @@ func (s *Session) firstNav(ctx context.Context) error {
 	return nil
 }
 
-// navToEnd waits for the page to be ready to receive scroll key events, by
-// trying to select an item with the right arrow key, and then scrolls down to the
-// end of the page, i.e. to the oldest items.
+// navToEnd selects the last item in the page
+//  by repeatedly advances the selected item with
+//  - kb.ArrowRight (which causes an initial selection, and/or advances it by one)
+//  - kb.End which scrolls to the end of the page, and advances the selected item.
+// Note timing is important, because when the kb.End causes significant scrolling,
+//  the active element become undefined for a certain time, in that case, we get an error (ignore), sleep, and retry.
+// The termnation criteria is that the selected item (document.activeElement.href) is stable for >2 iterations
 func navToEnd(ctx context.Context) error {
-	// wait for page to be loaded, i.e. that we can make an element active by using
-	// the right arrow key.
+	var prev, active string
+	lastRepeated := 0
 	for {
 		chromedp.KeyEvent(kb.ArrowRight).Do(ctx)
+		chromedp.KeyEvent(kb.End).Do(ctx)
 		time.Sleep(tick)
-		var ids []cdp.NodeID
-		if err := chromedp.Run(ctx,
-			chromedp.NodeIDs(`document.activeElement`, &ids, chromedp.ByJSPath)); err != nil {
-			return err
+
+		if err := chromedp.Evaluate(`document.activeElement.href`, &active).Do(ctx); err != nil {
+			time.Sleep(tick) // this extra sleep is important: after the kb.End, it sometimes takes a while for the active element to be reset
```

Comment:
Ok, for the comment, I will move to the line above. (I just pushed the update)

The reason the `continue` is preferable is that when the error condition occurs, (and enough time has passed, with the extra `Sleep`), the next thing that needs to happen is the `kb.ArrowRight`.

tl;dr

There are a lot of possible states. 
As far as I can tell (having experimented a lot last night):
- Initially, there is no active element, the ArrowRight will select the first photo. but the first `End` event does not actually perform the scroll until the ArrowRight has caused an element to be active (selected)
- On a subsequent iteration, the `End` event causes a scroll, and then there is *no* active element again (this is the tricky timing part)
  - if the scroll has stabilized (time-sensitive), then the next kb.ArrowRight will select the first photo at the new scroll position, (this allows advancement to continue from there.
  - however, if the scroll has not *stabilized*, the `ArrowRight` will have the effect of reselecting the first element back at the top of the page.
- Finally, after the `ArrowRight` has successfully selected an item at the new scroll position, all subsequent `kb.End` events have no effect.

So it turns that the way the loop is written is a pretty economical way of accomplishing everything we need. I did try to unroll every one of these steps, but it was less stable, and much more code.

Reply:
> Ok, for the comment, I will move to the line above. (I just pushed the update)
> 
> The reason the `continue` is preferable is that when the error condition occurs, (and enough time has passed, with the extra `Sleep`), the next thing that needs to happen is the `kb.ArrowRight`.
> 
> tl;dr
> 
> There are a lot of possible states.
> As far as I can tell (having experimented a lot last night):
> 
> * Initially, there is no active element, the ArrowRight will select the first photo. but the first `End` event does not actually perform the scroll until the ArrowRight has caused an element to be active (selected)
> * On a subsequent iteration, the `End` event causes a scroll, and then there is _no_ active element again (this is the tricky timing part)
>   
>   * if the scroll has stabilized (time-sensitive), then the next kb.ArrowRight will select the first photo at the new scroll position, (this allows advancement to continue from there.
>   * however, if the scroll has not _stabilized_, the `ArrowRight` will have the effect of reselecting the first element back at the top of the page.

I don't understand. Isn't it pretty bad to let that happen? Does this case still happen with the current patch?

> * Finally, after the `ArrowRight` has successfully selected an item at the new scroll position, all subsequent `kb.End` events have no effect.
> 
> So it turns that the way the loop is written is a pretty economical way of accomplishing everything we need. I did try to unroll every one of these steps, but it was less stable, and much more code.

Reply:
I believe you tested that part and it works, but I'm still confused about why and how it works, so I'll have to run a few experiments myself with it when I have a bit of time.

Reply:
FYI, I'm still experimenting on that part, because I don't like that we still rely on an arbitrary timeout.

Reply:
Ok, looking forward to your results.
I just pushed an update (See below)

Reply:
I found an algorithm that I think is simpler to understand (because it uses left arrow instead of right arrow), but it still suffers from needing one sleep, which sucks.

basically, what I want is:

```
	var first string
	for {
		chromedp.KeyEvent(kb.ArrowLeft).Do(ctx)
		if err := chromedp.Evaluate(`document.activeElement.href`, &first).Do(ctx); err != nil {
			continue
		}
		if first == "" {
			continue
		}
		break
	}

	var current string
	for {
		chromedp.KeyEvent(kb.End).Do(ctx)
		// TODO: get rid of that sleep
		time.Sleep(tick)
		for {
			chromedp.KeyEvent(kb.ArrowLeft).Do(ctx)
			if err := chromedp.Evaluate(`document.activeElement.href`, &current).Do(ctx); err != nil {
				continue
			}
			break
		}
		if current == first {
			continue
		}
		break
	}
```
And yes, I know that does not get us to the very last element. but I don't really care, that's easy to fix. Or we can just keep the call to navToLast after, whatever.

Isn't there anything that we can do after a call to 
    chromedp.KeyEvent(kb.End).Do(ctx)
that would allow us to actively detect and wait for the jump to be done?